sdk/storage/azblob/service/client.go

```diff
@@ -281,3 +281,40 @@ func (s *Client) FilterBlobs(ctx context.Context, where string, o *FilterBlobsOp
 	resp, err := s.generated().FilterBlobs(ctx, where, serviceFilterBlobsOptions)
 	return resp, err
 }
+
+// NewBatchBuilder creates an instance of BatchBuilder with the specified values.
+//   - cred - an Azure AD credential, typically obtained via the azidentity module
+//   - options - client options; pass nil to accept the default values
+func (s *Client) NewBatchBuilder(cred azcore.TokenCredential, options *ClientOptions) *BatchBuilder {
+	return nil
+}
+
+// NewBatchBuilderWithSharedKeyCredential creates an instance of BatchBuilder with the specified values.
+//   - cred - a SharedKeyCredential created with the matching container's storage account and access key
+//   - options - client options; pass nil to accept the default values
+func (s *Client) NewBatchBuilderWithSharedKeyCredential(cred *SharedKeyCredential, options *ClientOptions) *BatchBuilder {
+	return nil
+}
+
+// NewBatchBuilderWithNoCredential creates an instance of BatchBuilder with the specified values.
+// This is used to perform batch operations anonymously or with a shared access signature (SAS) token.
```

Comment:
Anonymous?

Reply:
Same as previous. This case can happen when public access is enabled on the storage account.
We will confirm this part while implementing this.

Reply:
Done. Renamed the method to `NewBatchBuilderWithSAS`.

Reply:
Same as container client.